### PR TITLE
Appt display sets 2

### PR DESCRIPTION
**Pull request by epsdky**
Added code to group appts with same date into sets for display in the Patient Summary
When the number of appts scheduled is greater than the number of appts configured to appear in the Appointments widget the last set turns green if all appts for the corresponding date are included for display and pink if all appts are not present.

### CODE REVIEW

File: interface/patient_file/summary/demographics.php [hunk 1]
```diff
@@ -1333,25 +1360,42 @@ function setMyPatient() {
...
+            echo htmlspecialchars(sprintf("%02d", $disphour) .":$dispmin " . xl($dispampm) . " (" . xl($dayname),ENT_NOQUOTES)  . ")</b> ";
             if ($row['pc_recurrtype']) echo "<img src='" . $GLOBALS['webroot'] . "/interface/main/calendar/modules/PostCalendar/pntemplates/default/images/repeating8.png' border='0' style='margin:0px 2px 0px 2px;' title='".htmlspecialchars(xl("Repeating event"),ENT_QUOTES)."' alt='".htmlspecialchars(xl("Repeating event"),ENT_QUOTES)."'>";
-            echo "<span title='" . generate_display_field(array('data_type'=>'1','list_id'=>'apptstat'),$row['pc_apptstatus']) . "'> ( " . htmlspecialchars($row['pc_apptstatus'],ENT_NOQUOTES) . " )</span>";
-            if ($row['pc_hometext']) echo "<font color='green'> CMT</font>";
-            echo "<br>" . htmlspecialchars(xl_appt_category($row['pc_catname']),ENT_NOQUOTES) . "<br>\n";
-            echo htmlspecialchars($row['ufname'] . " " . $row['ulname'],ENT_NOQUOTES) . "</a><br>\n";
+            echo "<span title='" . generate_display_field(array('data_type'=>'1','list_id'=>'apptstat'),$row['pc_apptstatus']) . "'> <br>Status( " . htmlspecialchars($row['pc_apptstatus'],ENT_NOQUOTES) . " ) </span>";
```
Comment (bradymiller) [inline interface/patient_file/summary/demographics.php]: need to translate Status via xlt function

File: interface/patient_file/summary/demographics.php [hunk 2]
```diff
@@ -1333,25 +1360,42 @@ function setMyPatient() {
...
+            echo htmlspecialchars(xl_appt_category($row['pc_catname']),ENT_NOQUOTES) . "\n";
+            if ($row['pc_hometext']) echo " <span style='color:green'> Com</span>";
+            echo "<br>" . htmlspecialchars($row['ufname'] . " " . $row['ulname'],ENT_NOQUOTES) . "</a></div>\n";
+            //////
```
Comment (olegsv) [inline interface/patient_file/summary/demographics.php]: This is very nice addition. I would've replaced inline styles with two classes, for example  "appts-more" and "appts-nomore" . This way the indicators' display can be customized via stylesheet without getting into code. 

I would've also add title attibute with a explanation.

Comment (epsdky) [inline interface/patient_file/summary/demographics.php]: Thanks Oleg.

**Comment by bradymiller** [comment]
Looks good. I committed this to the main codebase (I rebased the 3 commits into 1 commit). Thanks for yet another very nice contribution :)
-brady

**Pull request closed without merging**